crates/vqueues/src/scheduler/eligible.rs

```diff
@@ -149,17 +149,14 @@ impl EligibilityTracker {
                 State::NeedsPoll => {
                     let config = cache.config_pool().find(&qstate.qid.parent);
                     // update the state based on eligibility.
-                    match qstate
-                        .poll_eligibility(storage, SchedulerClock.now_millis(), meta, config)?
-                        .as_compact()
-                    {
+                    match qstate.poll_eligibility(storage, meta, config)?.as_compact() {
                         Eligibility::Eligible => {
                             *current_state = State::Ready;
                             return Ok(Some(handle));
                         }
                         Eligibility::EligibleAt(ts) => {
-                            let instant = SchedulerClock.millis_to_future_instant(ts);
-                            let timer_key = self.delayed_eligibility.insert_at(handle, instant);
+                            let duration = ts.duration_since(SchedulerClock.now_millis());
+                            let timer_key = self.delayed_eligibility.insert(handle, duration);
```

Comment:
The problems you've described in the commit description (Instant not being aware of/accounting for system sleeps/suspends), are they also affecting the `DelayQueue`? So can it happen that the `DelayQueue` returns items too late when calling `poll_next` if the system suspended/slept in between?